openbts/components.py

```diff
@@ -133,12 +213,39 @@ def delete_subscriber(self, imsi):
     response = self._send_and_receive(message)
     return response
 
-  def update_subscriber(self, new_name, new_msisdn, new_ipaddr, new_port, imsi):
+  def update_ipaddr(self, imsi, new_ipaddr):
+    """Updates a subscriber's IP address."""
+    message = {
+      'command': 'sip_buddies',
+      'action': 'update',
+      'match': {
+        'name': imsi
+      },
+      'fields': {
+        'ipaddr': new_ipaddr
+      }
+    }
+    return self._send_and_receive(message)
+
+  def update_port(self, imsi, new_port):
+    """Updates a subscriber's port."""
+    message = {
+      'command': 'sip_buddies',
+      'action': 'update',
+      'match': {
+        'name': imsi
+       },
+      'fields': {
+        'port': new_port,
+       }
+    }
+    return self._send_and_receive(message)
+
+  def update_subscriber(self, imsi, new_msisdn, new_ipaddr, new_port):
```

Comment:
Perhaps we could make these default to `None` and only update what's needed? This is assuming it's possible to do this update w/o requiring a lookup first.